src/snowflake/connector/result_batch.py

```diff
@@ -0,0 +1,549 @@
+    ):
+        super().__init__(
+            rowcount, chunk_headers, remote_chunk_info, column_names, use_dict_result
+        )
+        self._context = context
+        self._numpy = numpy
+        self._number_to_decimal = number_to_decimal
+
+    def __repr__(self) -> str:
+        return f"ArrowResultChunk({self.rowcount})"
+
+    def _load(
+        self, response, row_unit: str
+    ) -> Union[Iterator[Union[Dict, Exception]], Iterator[Union[Tuple, Exception]]]:
+        """Creates a `PyArrowIterator` from a response.
+
+        This is used to iterate through results in different ways depending on which
+        mode that `PyArrowIterator` is in.
+        """
+        from .arrow_iterator import PyArrowIterator
+
+        gfd = GzipFile(fileobj=response.raw, mode="r")
+
+        iter = PyArrowIterator(
+            None,
+            gfd,
+            self._context,
+            self._use_dict_result,
+            self._numpy,
+            self._number_to_decimal,
+        )
+        if row_unit == TABLE_UNIT:
+            iter.init_table_unit()
+        return iter
+
+    def _from_data(
+        self, data: str, iter_unit: str
+    ) -> Union[Iterator[Union[Dict, Exception]], Iterator[Union[Tuple, Exception]]]:
+        """Creates a `PyArrowIterator` files from a str.
+
+        This is used to iterate through results in different ways depending on which
+        mode that `PyArrowIterator` is in.
+        """
+        from .arrow_iterator import PyArrowIterator
+
+        if len(data) == 0:
+            return iter([])
+
+        _iter = PyArrowIterator(
+            None,
+            io.BytesIO(b64decode(data)),
+            self._context,
+            self._use_dict_result,
+            self._numpy,
+            self._number_to_decimal,
+        )
+        if iter_unit == TABLE_UNIT:
+            _iter.init_table_unit()
+        else:
+            _iter.init_row_unit()
+        return _iter
+
+    @classmethod
+    def from_data(
+        cls,
+        data: str,
+        context: "ArrowConverterContext",
+        use_dict_result: bool,
+        numpy: bool,
+        column_names: Sequence[str],
+        number_to_decimal: bool,
+    ):
+        """Initializes an `ArrowResultBatch` from static, local data."""
+        new_chunk = cls(
+            len(data),
+            None,
+            None,
+            context,
+            use_dict_result,
+            numpy,
+            column_names,
+            number_to_decimal,
+        )
+        new_chunk._data = data
+        new_chunk.rowcount = len(list(new_chunk._from_data(data, ROW_UNIT)))
+        return new_chunk
+
+    def _download(
+        self, **kwargs
+    ) -> Union[
+        Iterator[Union[Dict, Exception]],
+        Iterator[Union[Tuple, Exception]],
+        Iterator[Table],
+    ]:
+        iter_unit = kwargs.pop("iter_unit", ROW_UNIT)
+        if self._local:
+            return self._from_data(self._data, iter_unit)
+        sleep_timer = 1
```

Comment:
next 30 lines are the same as the code in JsonResultBatch._download and might be able to use a common method

Reply:
I de-duplicated the code snippets